examples/marquee.lisp

```diff
@@ -0,0 +1,88 @@
+(defpackage #:charms-marquee
+  (:use #:cl)
+  (:export #:main))
+(in-package :charms-marquee)
+
+(defclass marquee ()
+  ((position :initarg :position :accessor marquee-position)
+   (size :initarg :size :accessor marquee-size)
+   (text :initarg :text :accessor marquee-text)
+   (viewport :initarg :viewport :reader marquee-viewport)
+   (padding-char :initarg :padding-char :initform #\Space :accessor marquee-padding-char)
+   (buffer :initform "" :accessor marquee-buffer))
+  (:documentation "A marquee is like a scrolling text thing. A string of characters moves through
+a viewport. The viewport constrains the text so you only see a little bit at a time."))
+
+;; (defmethod marquee-viewport :before ((marquee marquee))
+;;   "Before getting the MARQUEE viewport (the text that is shown at a given time), the buffer
+;; needs to be padded in case the text doesn't fill the MARQUEE size"
+;;  ( )
+
+(defmethod marquee-viewport ((marquee marquee))
+  "Evaluates to the text currently shown in the viewport for the given MARQUEE."
+  (marquee-pad-buffer marquee)
+  (subseq (slot-value marquee 'buffer) 0 (slot-value marquee 'size)))
```

Comment:
use MARQUEE-BUFFER and MARQUEE-SIZE instead of SLOT-VALUE.

SLOT-VALUE in general should be reserved for lower level and/or surgical operations on classes that skirt the reader/accessor method logic.